git/repo/base.py

```diff
@@ -124,6 +124,8 @@ def __init__(self, path=None, odbt=DefaultDBType, search_parent_directories=Fals
                 repo = Repo("~/Development/git-python.git")
                 repo = Repo("$REPOSITORIES/Development/git-python.git")
 
+            if `None, current-directory is used.
+            The :envvar:`GIT_DIR` if set and not empty takes precendance over this parameter.
```

Comment:
The environment variable taking precedence over the specified path seems error prone and undesirable. For instance, with `GIT_DIR` set, you can no longer do something like this:

```
repo_1 = git.Repo('/path/to/repo1')
repo_2 = git.Repo('/path/to/repo2')
```
I think `GIT_DIR` as a fallback makes sense (if `path` is `None` for instance). However, is there any reason to prefer it if an explicit path is provided?